Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/NativeLibrary.java

```diff
@@ -401,18 +406,64 @@ private NativeLibrary()
 		CacheClassesAndMethods();
 	}
 
-	public static void displayAlertMsg(final String alert)
+	public static boolean displayAlertMsg(final String caption, final String text, final boolean yesNo)
 	{
-		Log.error("[NativeLibrary] Alert: " + alert);
+		Log.error("[NativeLibrary] Alert: " + text);
 		final EmulationActivity emulationActivity = sEmulationActivity.get();
-		if (emulationActivity != null)
+		boolean result = false;
+		if (emulationActivity == null)
 		{
-			emulationActivity.runOnUiThread(() -> Toast.makeText(emulationActivity, "Panic Alert: " + alert, Toast.LENGTH_LONG).show());
+			Log.warning("[NativeLibrary] EmulationActivity is null, can't do panic alert.");
+		}
+		else if (yesNo)
+		{
+			result = displayYesNo(emulationActivity, caption, text);
 		}
 		else
 		{
-			Log.warning("[NativeLibrary] EmulationActivity is null, can't do panic toast.");
+			emulationActivity.runOnUiThread(() -> Toast.makeText(emulationActivity, "Panic Alert: " + text, Toast.LENGTH_LONG).show());
```

Comment:
Could we make this one an AlertDialog too instead of it remaining a Toast? Toasts have the problem that you don't have as much time as you want to for reading them, which is a problem because many of the users don't speak English as their native language and because some messages are pretty long (like the unknown FIFO opcode one).

Reply:
Yes, changed the code to allow for that, see [commit #d6dee63](https://github.com/dolphin-emu/dolphin/pull/6295/commits/d6dee635d4f07a022c9614fb0284ccd3fb931dea).